calico_kubernetes/calico_kubernetes.py

```diff
@@ -228,15 +299,30 @@ def _container_remove(self, hostname, orchestrator_id):
             sys.exit(1)
 
         # Remove any IP address assignments that this endpoint has
+        ip_set = set()
         for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
             assert(net.size == 1)
-            self._datastore_client.unassign_address(None, net.ip)
+            ip_set.add(net.ip)
 
-        # Remove the endpoint
-        netns.remove_veth(endpoint.name)
+        logger.info("Removing IP addresses %s from endpoint %s", ip_set, endpoint.name)
+        self._datastore_client.release_ips(ip_set)
 
-        # Remove the container from the datastore.
-        self._datastore_client.remove_workload(hostname, orchestrator_id, self.docker_id)
+        # Remove the veth interface from endpoint
+        if netns.remove_veth(endpoint.name):
```

Comment:
remove_veth(...) doesn't return False when it hits an error - it raises a CalledProcessError.  It will return false if the veth doesn't exist.  From our perspective, we don't care about this case (since we wan't the veth to not exist anyway).

Also - In general, it's usually not a good idea to add extra branches just for logging.

I might do the following instead:
```
_log.info("Removing veth interface from endpoint %s" % endpoint.name)
try:
  netns.remove_veth(endpoint.name)
except:
  _log.exception(...)
```